src/styles/getTooltipStyles.js

```diff
@@ -0,0 +1,176 @@
+    const componentCenter = xOffset + (componentWidth / 2);
+    const tooltipLeftEdge = componentCenter - (tooltipWidth / 2);
+    const tooltipRightEdge = componentCenter + (tooltipWidth / 2);
+
+    if (tooltipLeftEdge < GUTTER_WIDTH) {
+        // Tooltip is in left gutter, shift right by a multiple of four.
+        return roundToNearestMultipleOfFour(GUTTER_WIDTH - tooltipLeftEdge);
+    }
+
+    if (tooltipRightEdge > (windowWidth - GUTTER_WIDTH)) {
+        // Tooltip is in right gutter, shift left by a multiple of four.
+        return roundToNearestMultipleOfFour(windowWidth - GUTTER_WIDTH - tooltipRightEdge);
+    }
+
+    // Tooltip is not in the gutter, so no need to shift it horizontally
+    return 0;
+}
+
+/**
+ * Generate styles for the tooltip component.
+ *
+ * @param {Number} interpolatedSize - The current size of the tooltip used in the scaling animation.
+ * @param {Number} windowWidth - The width of the window.
+ * @param {Number} xOffset - The distance between the left edge of the window
+ *                           and the left edge of the wrapped component.
+ * @param {Number} yOffset - The distance between the top edge of the window
+ *                           and the top edge of the wrapped component.
+ * @param {Number} componentWidth - The width of the wrapped component.
+ * @param {Number} componentHeight - The height of the wrapped component.
+ * @param {Number} tooltipWidth - The width of the tooltip itself.
+ * @param {Number} tooltipHeight - The height of the tooltip itself.
+ * @returns {Object}
+ */
+export default function getTooltipStyles(
+    interpolatedSize,
+    windowWidth,
+    xOffset,
+    yOffset,
+    componentWidth,
+    componentHeight,
+    tooltipWidth,
+    tooltipHeight,
+) {
+    // Determine if the tooltip should display below the wrapped component.
+    // If a tooltip will try to render within GUTTER_WIDTH logical pixels of the top of the screen,
+    // we'll display it beneath its wrapped component rather than above it as usual.
+    const shouldShowBelow = (yOffset - tooltipHeight) < GUTTER_WIDTH;
+
+    // Determine if we need to shift the tooltip horizontally to prevent it
+    // from displaying too near to the edge of the screen.
+    const horizontalShift = computeHorizontalShift(windowWidth, xOffset, componentWidth, tooltipWidth);
+
+    const tooltipVerticalPadding = spacing.pv1;
+    const tooltipFontSize = variables.fontSizeSmall;
+
+    return {
+        animationStyle: {
+            transform: [{
+                scale: interpolatedSize,
+            }],
+        },
+        tooltipWrapperStyle: {
+            position: 'absolute',
+            backgroundColor: themeColors.heading,
+            borderRadius: variables.componentBorderRadiusSmall,
+            ...tooltipVerticalPadding,
+            ...spacing.ph2,
+
+            // Shift the tooltip down (+) by...
+            //
+            //   component height
+            // + (pointer height - 1)
```

Comment:
Why `1` and not `4` ?

Reply:
By default, the tooltip top is at the top of the component. So first it's shifted down by the component height so that the tooltip top is at the component bottom. Then the pointer is in the way, but if you shift it down by the height of the pointer, `4`, then there is a small but noticeable gap (presumably 1px) between the tooltip and the pointer. If you shift it down by `3` it looks right.